src/src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -705,42 +705,174 @@ void VM_Version::get_processor_features() {
       _features &= ~CPU_VZEROUPPER;
     }
   }
-
+  enum features { cmov, cx8, fxsr, mmx, sse, sse2, sse3, ssse3, sse41, sse42, popcnt, avx, avx2, aes, clmul, erms, rtm, mmxext, dnowperf3, lzcnt, sse4a, ht, tsc, tscinvbit, tscinv, bmi1, bmi2, adx, evex, sha, fma};
+  char result[256];
   char buf[256];
-  jio_snprintf(buf, sizeof(buf), "(%u cores per cpu, %u threads per core) family %d model %d stepping %d%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s%s",
+  for(features fea=cmov ; fea<=fma; fea=(features)(fea+1)){
```

Comment:
you shouldn't hard-code cmov or fma here.  you can define a sentinel enum such as feature_numbers=fma+1.

for (int i=0; i< feature_numbers; ++i)